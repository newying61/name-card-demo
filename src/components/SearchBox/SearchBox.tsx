import React from 'react';
import Form from 'react-bootstrap/Form';
import { FormInput } from '../Styled/FormInput';

interface SearchBoxProps {
  searchTerm: string;
  handleChange: (val: string) => void;
}

const SearchBox: React.FC<SearchBoxProps> = ({ searchTerm, handleChange }) => {
  const onChange = (e: any) => {
    const val = e.target.value;
    handleChange(val)
  }

  return (
    <Form.Group controlId="search-box">
      <Form.Label>Search by name</Form.Label>
      <FormInput
        type="text"
        placeholder=""
        name="searchBox"
        value={searchTerm}
        onChange={onChange}
      />
    </Form.Group>
  );
}

export default SearchBox;
