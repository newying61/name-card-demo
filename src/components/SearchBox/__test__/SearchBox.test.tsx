import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import SearchBox from '../';

afterEach(cleanup);

describe('SearchBox', () => {
  it('should render without crash', () => {
    const onChange = jest.fn();

    const { container } = render(
      <SearchBox searchTerm="test" handleChange={onChange} />
    );

    fireEvent.change(container.querySelector('#search-box'), { target: { value: 'a' } });

    expect(onChange).toBeCalledWith('a');

    expect(container.firstChild).toMatchSnapshot();
  });
});