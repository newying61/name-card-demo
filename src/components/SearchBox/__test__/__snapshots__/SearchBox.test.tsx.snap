// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox should render without crash 1`] = `
<div
  class="form-group"
>
  <label
    class="form-label"
    for="search-box"
  >
    Search by name
  </label>
  <input
    class="sc-bdVaJa btHEdo form-control"
    id="search-box"
    name="searchBox"
    placeholder=""
    type="text"
    value="test"
  />
</div>
`;
